tests/field/test_selffieldforces.py

```diff
@@ -205,37 +205,59 @@ def test_update_points(self):
 
         # This test is incomplete.
 
-    def test_forces_taylor_test(self):
+    def test_meansquaredforces_taylor_test(self):
         """Verify that dJ matches finite differences of J"""
         # The Fourier spectrum of the NCSX coils is truncated - we don't need the
         # actual coil shapes from the experiment, just a few nonzero dofs.
-        curves, currents, axis = get_ncsx_data(Nt_coils=2)
 
-        # The next line is needed temporarily because ForceOpt doesn't yet
-        # account for derivatives with respect to currents. This next line
-        # should be removed once that derivative is added.
+        curves, currents, axis = get_ncsx_data(Nt_coils=2)
+        coils = [Coil(curve, current) for curve, current in zip(curves, currents)]
         [current.fix_all() for current in currents]
 
+        J = MeanSquaredForce(coils[0], coils, regularization_circ(0.05))
+        dJ = J.dJ()
+        deriv = np.sum(dJ * np.ones_like(J.x))
+        dofs = J.x
+        h = np.ones_like(dofs)
+        err = 100
+        for i in range(10, 19):
+            eps = 0.5**i
+            J.x = dofs + eps * h
+            Jp = J.J()
+            J.x = dofs - eps * h
+            Jm = J.J()
+            deriv_est = (Jp - Jm) / (2 * eps)
+            err_new = np.abs(deriv_est - deriv) / np.abs(deriv)
+            # print("i:", i, "deriv_est:", deriv_est, "deriv:", deriv, "err_new:", err_new, "err:", err, "ratio:", err_new / err)
+            np.testing.assert_array_less(err_new, 0.6 * err)
```

Comment:
Since the finite difference step size is being reduced by 1/2 each iteration, and centered differences are being used, we'd expect the error to go down by 1/4 each iteration. That factor of 1/4 was rounded up to 0.3 in other Taylor tests to provide a bit of margin. If you're seeing this factor needs to be as large as 0.6, that may indicate an error in the derivative? Not sure, but worth looking into.

Reply:
The problem is as I thought, the error is around 0.25 until the final iteration, so we probably just encountered roundoff error there.